info.limpet.rcp/src/info/limpet/rcp/data_provider/data/ReflectivePropertySource.java

```diff
@@ -0,0 +1,174 @@
+
+	@Override
+	public IPropertyDescriptor[] getPropertyDescriptors()
+	{
+		if (propertyDescriptors == null)
+		{
+			initPropertyDescriptors(object.getClass());
+		}
+		return propertyDescriptors;
+	}
+
+	private void initPropertyDescriptors(Class<?> cls)
+	{
+
+		descriptorPerProperty = new HashMap<String, PropertyDescriptor>();
+
+		List<IPropertyDescriptor> result = new ArrayList<IPropertyDescriptor>();
+
+		try
+		{
+			PropertyDescriptor[] beanPropertyDescriptors = Introspector.getBeanInfo(
+					cls).getPropertyDescriptors();
+			for (PropertyDescriptor beanPropertyDescriptor : beanPropertyDescriptors)
+			{
+				UIProperty annotation = beanPropertyDescriptor.getReadMethod()
+						.getAnnotation(UIProperty.class);
+				if (annotation != null)
+				{
+
+					org.eclipse.ui.views.properties.PropertyDescriptor descriptor = null;
+					
+					String propId = beanPropertyDescriptor.getName();
+					String propName = annotation.name();
+					String propCategory = annotation.category();
+					
+					if (beanPropertyDescriptor.getWriteMethod() != null)
+					{
+						Class<?> propertyType = beanPropertyDescriptor.getPropertyType();
+						if (propertyType.equals(String.class))
+						{
+							descriptor = new TextPropertyDescriptor(
+									propId, propName);
+						}
+						else if ("boolean".equals(propertyType.getName()))
+						{
+							descriptor = new CheckboxPropertyDescriptor(
+									propId, propName);
+						}
+						else if ("int".equals(propertyType.getName()))
+						{
+							int min = annotation.min();
+							int max = annotation.max();
+							
+							descriptor =  new SliderPropertyDescriptor(
+									propId, propName, min, max);
+						}						
+						else
+						{
+							// TODO: handle other property types, such as
+							// number, complex types, Wrapper types - Boolean, Integer, etc
+						}
+					}
+					else
+					{
+						// read only descriptor
+						descriptor = new org.eclipse.ui.views.properties.PropertyDescriptor(
+								beanPropertyDescriptor.getName(), propName);
+					}
+
+					if (descriptor != null)
+					{
+						descriptor.setCategory(propCategory);
+						result.add(descriptor);
+
+						descriptorPerProperty.put(beanPropertyDescriptor.getName(),
+								beanPropertyDescriptor);
+					}
+
+				}
+			}
+		}
+		catch (IntrospectionException e)
+		{
+			Activator.logError(Status.ERROR,
+					"Could not load property descriptors for class " + cls.getName(), e);
+
+		}
+
+		propertyDescriptors = result
+				.toArray(new IPropertyDescriptor[result.size()]);
+	}
+
+	@Override
+	public Object getPropertyValue(Object id)
+	{
+		PropertyDescriptor descriptor = descriptorPerProperty.get(id);
+		try
+		{
+			return descriptor.getReadMethod().invoke(object);
+		}
+		catch (Exception e)
+		{
+			Activator.logError(Status.ERROR, "Could not retrive value for property "
+					+ id, e);
+			return null;
+		}
+	}
+
+	@Override
+	public boolean isPropertySet(Object id)
+	{
+		// TODO Auto-generated method stub
```

Comment:
@dinkoivanov - could you have a go at implementing isPropertySet and resetPropertyValue ?